kpi/mixins/export_object.py

```diff
@@ -70,13 +70,7 @@ def get_export_object(
             PERM_VIEW_SUBMISSIONS not in source_perms
             and PERM_PARTIAL_SUBMISSIONS not in source_perms
         ):
-            # Unsure if DRF exceptions make sense here since we're not
-            # returning a HTTP response
```

Comment:
i think the comment is still valid: `get_export_object()` is called by a celery task as well as your new view.

my proposal would be to:
1. create a new exception class for this kind of permission-denied-during-export error
1. raise the exception here
1. catch the exception in the view and have the view return a 404
1. in the celery task, add a more specific exception handler [ref](https://github.com/kobotoolbox/kpi/blob/c2bb9d3066d6523f270df09283ecf62b4af21709/kpi/models/import_export_task.py#L110-L114) that catches this new exception, marks the job as failed, and sets an appropriate error message

i'll write a draft of this and ask you to review it

Reply:
https://github.com/kobotoolbox/kpi/pull/3713